cfbs/masterfiles/generate_vcf_git_checkout.py

```diff
@@ -0,0 +1,113 @@
+import os
+import shutil
+import subprocess
+import sys
+
+from cfbs.utils import write_json
+from cfbs.masterfiles.analyze import (
+    finalize_vcf,
+    initialize_vcf,
+    versions_checksums_files,
+)
+
+DIR_PATH = "."
+"""The path of the working directory."""
+
+MPF_URL = "https://github.com/cfengine/masterfiles"
+MPF_PATH = os.path.join(DIR_PATH, "masterfiles")
+
+
+def check_required_command(command):
+    if not shutil.which(command):
+        print("`%s` was not found" % command)
+        sys.exit(1)
+
+
+def check_required_commands(commands):
+    for c in commands:
+        check_required_command(c)
+
+
+def generate_vcf_git_checkout(checkout_tags):
+    required_commands = ["git", "make", "automake", "autoconf"]
+    check_required_commands(required_commands)
+
+    # get the current version of the MPF repo
+    if not os.path.isdir(MPF_PATH):
+        subprocess.run(
+            ["git", "clone", "--no-checkout", MPF_URL],
+            cwd=DIR_PATH,
+            check=True,
+        )
+    else:
+        subprocess.run(
+            ["git", "fetch", "--all"],
+            cwd=MPF_PATH,
+            check=True,
+        )
+
+    versions_dict, checksums_dict, files_dict = initialize_vcf()
+
+    for tag in checkout_tags:
+        print("Checkouting tag", tag)
+
+        # checkouting some tags equal to the downloaded version doesn't result in the same files
+        # the downloadable files are reproducible by checkouting specific tags
```

Comment:
I might suggest that we FIX these tags that are exceptions here. It would seem of not much consequence since all of these versions are out of support.